File: kolibri-swift/Examples/handshake.swift
// A minimal handshake example, mirroring kolibri-go/example/handshake.
//
// Not built by the package. To run it, add an executable target to Package.swift
// (or paste the body into your own app) after building the native lib with
// ./build-rust.sh.
import Kolibri

@main
struct Handshake {
    static func main() async {
        var config = Config(host: "api.oneme.ru")
        config.onWire = { event in
            let arrow = event.direction == "out" ? "->" : "<-"
            let json = event.json.prefix(90)
            print("\(arrow) \(event.command) op=\(event.opcode) seq=\(event.seq) \(json)")
        }

        do {
            let session = try Session(config: config)

            let info = try await session.connect()
            print("connected: state=\(session.state), callsSeed=\(info["callsSeed"] ?? "nil")")

            print("switching interactive=false on the live socket")
            session.pingInteractive = false

            session.disconnect()
        } catch {
            print("error: \(error)")
        }
    }
}

File: kolibri-kotlin/example/src/main/kotlin/ru/kolibri/example/Handshake.kt
package ru.kolibri.example

import kotlinx.coroutines.flow.take
import kotlinx.coroutines.runBlocking
import ru.kolibri.Config
import ru.kolibri.Session
import ru.kolibri.WireEvent

/**
 * Connects, prints the handshake, sends one request, and drains a few pushes.
 *
 * Run with:  gradle :example:run --args "api.oneme.ru"
 * (build the native lib first: ./build-rust.sh)
 */
fun main(args: Array<String>) = runBlocking {
    val host = args.firstOrNull() ?: "api.oneme.ru"

    val config = Config(
        host = host,
        onWire = { e: WireEvent ->
            println("${e.direction} ${e.command} op=${e.opcode} ${e.json}")
        },
    )

    Session.open(config).use { session ->
        val info = session.connect()
        println("handshake -> $info")
        println("state=${session.state} ua=${session.userAgent}")

        // JSON in, JSON out; no msgpack library needed.
        val resp = session.requestJson(opcode = 64, jsonIn = "{}")
        println("response -> $resp")

        session.pushes().take(3).collect { push ->
            println("push op=${push.opcode} ${push.json}")
        }
    }
}
